telemetry-to-azure-iot-hub/main.bicep

```diff
@@ -0,0 +1,193 @@
+// Azure Bicep template deploying the required resources to send telemetry from
+// an Axis camera to Azure IoT Hub.
+//
+// The template can be deployed from a terminal using the following command.
+//
+//   az deployment group create -g <resource group> -f main.bicep --parameters \
+//     objectId=$(az ad signed-in-user show --query objectId --output tsv) \
+//     organizationName=<organization name> \
+//     iotHubName=<IoT Hub name> \
+//     deviceIdentity=<IoT device identity>
+
+@description('''
+  The object ID of your Azure Active Directory user. The object ID can be found
+  either by navigating to your user in Azure Active Directory, or by running the
+  following command in the Azure Cloud Shell:
+  'az ad signed-in-user show --query objectId --output tsv'.
+  ''')
+param objectId string
+
+@description('''
+  The name of your organization, used when generating the X.509 certificates.
+  ''')
+param organizationName string
+
+@description('''
+  The name of the IoT Hub. Please note that the name must be unique, i.e. two
+  IoT Hubs in Azure cannot share the same name.
+  ''')
+param iotHubName string = 'axis-telemetry-${uniqueString(resourceGroup().id)}'
+
+@description('''
+  The name of the IoT device, used for authentication and access control.
+  ''')
+param deviceIdentity string = 'device01'
+
+var location = resourceGroup().location
+var tenantId = subscription().tenantId
+
+// -----------------------------------------------------------------------------
+// Azure IoT Hub receiving telemetry from the Axis camera
+// -----------------------------------------------------------------------------
+
+resource iotHub 'Microsoft.Devices/IotHubs@2021-03-31' = {
+  name: iotHubName
+  location: location
+  sku: {
+    name: 'S1'
+    capacity: 1
+  }
+}
+
+// -----------------------------------------------------------------------------
+// User responsible for generating X.509 certificates and uploading them to
+// Azure Key Vault
+// -----------------------------------------------------------------------------
+
+resource identity 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
+  name: 'telemetry-to-azure-iot-hub'
+  location: location
+}
+
+resource roleAssignment 'Microsoft.Authorization/roleAssignments@2020-08-01-preview' = {
+  name: guid(resourceGroup().id, 'contributor')
+  properties: {
+    principalId: identity.properties.principalId
+    roleDefinitionId: '${subscription().id}/providers/Microsoft.Authorization/roleDefinitions/b24988ac-6180-42a0-ab88-20f7382dd24c'
```

Comment:
Should we add a reference to where this built-in role-id comes from? Similar to in the images-to-azure-storage example.